pytorch_lightning/utilities/grads.py

```diff
@@ -35,18 +35,13 @@ def grad_norm(module: Module, norm_type: Union[float, int, str]) -> Dict[str, fl
             as a single vector.
     """
     norm_type = float(norm_type)
-
-    norms, all_norms = {}, []
-    for name, p in module.named_parameters():
-        if p.grad is None:
-            continue
-
-        param_norm = float(p.grad.data.norm(norm_type))
```

Comment:
Improvement: Use `.item()` over `float()`

Reply:
`.item()` will trigger a device to host copy which would incur a perf hit

Reply:
Same as for `float()`, but `.item()` does not impose the type conversion